src/tools/base/BaseTool.js

```diff
@@ -17,20 +18,28 @@ import mixins from './../../mixins/index.js';
 class BaseTool {
   /**
    * Constructor description
-   * @param {ToolConfiguration} [ToolConfiguration={}]
+   * @param {defaultProps} [defaultProps={}] Tools Default properties
+   * @param {props} [props={}] Tool properties set on instantiation of a tool
    */
-  constructor({
-    name,
-    strategies,
-    defaultStrategy,
-    configuration,
-    supportedInteractionTypes,
-    mixins,
-  } = {}) {
+  constructor(props, defaultProps) {
+    /**
+     * Merge default props with custom props
+     */
+    props = lodash.merge({}, defaultProps, props);
+
+    const {
+      name,
+      strategies,
+      defaultStrategy,
+      configuration,
+      supportedInteractionTypes,
+      mixins,
+    } = props;
```

Comment:
I'm having a hard time remembering. Does this set the object's properties because the variables were instantiated here? Or do we need explicitly use `this.`? For clarity's sake, we may want to use `this`

Reply:
It does not. and also we do it one by one because we do some checks, as: 
- if we don't have any `defaultStrategy` we set the first. 
- if we don't have `strategies` we set as an empty array
...